mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.js

```diff
@@ -1,36 +1,59 @@
-import { Checkbox, FormControlLabel, FormGroup, TextField } from '@mui/material'
+import { Checkbox, FormControlLabel, FormGroup, Stack, TextField } from '@mui/material'
 import PropTypes from 'prop-types'
-import { forwardRef, useEffect, useState } from 'react'
+import { forwardRef, useState } from 'react'
 import language from '../../language'
+import { makeValidClassName } from '../../library/strings/makeValidClassName'
 
-const getValueToReturn = ({ nonOtherSelectedValues, otherInputValue, isOtherChecked }) => {
+const getValueToReturn = ({
+  nonOtherSelectedValues,
+  otherInputValue,
+  isOtherChecked,
+  event,
+  shouldReturnEvent
+}) => {
   const valueToReturn = { selectedValues: nonOtherSelectedValues ?? [] }
+  valueToReturn.isOtherChecked = !!isOtherChecked //we're being explicit here for the sake of composing code not having to use empty strings vs undefined to differentiate between checked and unchecked for validation for example.
   if (isOtherChecked) {
     valueToReturn.otherValue = otherInputValue
-    valueToReturn.isOtherChecked = true //we're being explicit here for the sake of composing code not having to use empty strings vs undefined to differentiate between checked and unchecked for validation for example.
+  }
+  if (shouldReturnEvent) {
+    // react hook form doesnt like the event being attached to the value object if it handles onchange, so made a flag for when we are handling onchange manually and want to use the event
+    valueToReturn.event = event
```

Comment:
fixed bug I found.

Reply:
Im merging now, but please ping me in slack if you have any comments on this bugfix.